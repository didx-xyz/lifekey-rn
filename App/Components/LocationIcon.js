
import React from "react"
import Svg, { Rect, Polygon } from "react-native-svg"
import PropTypes from "prop-types"

const LocationIcon = (props) => {
  const dynamic = {
    "stroke": props.stroke
  }

  return (
    <Svg width={props.width} height={props.height} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 23.4 18.4">
      <Rect {...styles.cls1} {...dynamic} x="0.5" y="0.5" width="22.4" height="17.3"/>
      <Polygon {...styles.cls1} {...dynamic} points="6.9,9.2 0.5,15.6 0.5,17.9 3.9,17.9 10.4,11.3 22.9,11.3 22.9,7.1 10.4,7.1 3.9,0.5 0.5,0.5 0.5,2.8 6.9,9.2"/>
    </Svg>
  )
}

LocationIcon.defaultProps = {
  "stroke": "#000"
}

LocationIcon.propTypes = {
  "width": PropTypes.number.isRequired,
  "height": PropTypes.number.isRequired,
  "stroke": PropTypes.string
}

const styles = {
  "cls1": {
    "fill": "none",
    "strokeLinecap": "round",
    "strokeLinejoin": "round"
  }
}

export default LocationIcon
