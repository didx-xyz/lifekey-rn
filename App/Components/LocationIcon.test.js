// external dependencies
import React from "react"

// internal dependencies
import LocationIcon from "./LocationIcon"

test("LocationIcon renders correctly, by default", function() {
  const tree = shallow(
    <LocationIcon width={16} height={16} />
  )

  expect(tree).toMatchSnapshot()
})

test("LocationIcon renders correctly, when modified", function() {
  const tree = shallow(
    <LocationIcon width={32} height={32} stroke="#fff" />
  )

  expect(tree).toMatchSnapshot()
})
