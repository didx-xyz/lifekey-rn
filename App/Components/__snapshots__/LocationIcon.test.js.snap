// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationIcon renders correctly, by default 1`] = `
<Svg
  height={16}
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 23.4 18.4"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <Rect
    fill="none"
    height="17.3"
    rx={0}
    ry={0}
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
    width="22.4"
    x="0.5"
    y="0.5"
  />
  <Polygon
    fill="none"
    points="6.9,9.2 0.5,15.6 0.5,17.9 3.9,17.9 10.4,11.3 22.9,11.3 22.9,7.1 10.4,7.1 3.9,0.5 0.5,0.5 0.5,2.8 6.9,9.2"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</Svg>
`;

exports[`LocationIcon renders correctly, when modified 1`] = `
<Svg
  height={32}
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 23.4 18.4"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <Rect
    fill="none"
    height="17.3"
    rx={0}
    ry={0}
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
    width="22.4"
    x="0.5"
    y="0.5"
  />
  <Polygon
    fill="none"
    points="6.9,9.2 0.5,15.6 0.5,17.9 3.9,17.9 10.4,11.3 22.9,11.3 22.9,7.1 10.4,7.1 3.9,0.5 0.5,0.5 0.5,2.8 6.9,9.2"
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</Svg>
`;
